src/sentry/api/endpoints/group_similar_issues_embeddings.py

```diff
@@ -111,10 +140,13 @@ def get(self, request: Request, group) -> Response:
         latest_event = group.get_latest_event()
         stacktrace_string = ""
         if latest_event.data.get("exception"):
-            stacktrace_string = get_stacktrace_string(latest_event.data["exception"], latest_event)
+            grouping_info_response = EventGroupingInfoEndpoint().get(
```

Comment:
can the code be refactored in such a way where we're not calling an endpoint like this? Or is there a very good reason why you're doing it this way?

Reply:
Thanks for the feedback! Is there a better way to call an endpoint? The reason I'm doing this is because we want to use this endpoint to get the frames used for grouping.

Alternatively, I could move the `/grouping-info/` endpoint logic into a helper function and just call the helper function instead of the endpoint.

Reply:
yes, you basically never want to call an endpoint directly from your own code. that is because endpoint methods are meant to be called by the URL router / the django framework. (you had to hack a mock `HttpRequest`, and there are all sorts of other issues with doing it this way, like auth/security). they aren't functions for you to call, generally. moving logic into a helper function here is recommended.